### PR TITLE
Add support for max connections

### PR DESCRIPTION
**Pull request by rikatz**
Signed-off-by: Ricardo Pchevuzinske Katz <ricardo.katz@gmail.com>

Go netutils add support for max connection.

If you surpass this limit, it will be queued by Go until the connection can be processed OR gets a timeout

### CODE REVIEW

**Comment by cpaillet** [comment]
Ok for me, can you just rebase libs ?

**Comment by rikatz** [comment]
Sure :) Doing right now

**Comment by rikatz** [comment]
@cpaillet sorry, gofmt changed some other stuff, can revert here if you prefer (only tabs, etc

**Comment by cpaillet** [comment]
> 
> 
> @cpaillet sorry, gofmt changed some other stuff, can revert here if you prefer (only tabs, etc

Hi,

sorry, my fault. I merge another review to fix that. ( And I just add a linter in the tests )
Next time you can split your PR in 2 review, it's more readable.

**Review by cpaillet — APPROVED** [review]

File: spoe.go [hunk 1]
```diff
@@ -67,6 +70,11 @@ func (a *Agent) ListenAndServe(addr string) error {
 	if err != nil {
 		return errors.Wrap(err, "spoe")
 	}
+	defer lis.Close()
+	if a.cfg.MaxConnections > 0 {
+		log.Infof("spoe: max connections: %d", a.cfg.MaxConnections)
+		return a.Serve(netutil.LimitListener(lis, a.cfg.MaxConnections))
```
Comment (ShimmerGlass) [inline spoe.go]:
```suggestion
		lis = netutil.LimitListener(lis, a.cfg.MaxConnections)
```

**Review by ShimmerGlass — CHANGES_REQUESTED** [review]
Looks good to met. Left a little style suggestion to simplify things.

**Comment by rikatz** [comment]
@ShimmerGlass thanks, commited your proposal :)

**Comment by ShimmerGlass** [comment]
Thank you for your contribution!

**Pull request merged**